### DISCUSSION TITLE
Rapid v2.0 WebGL Announcement

### DISCUSSION THREAD

**Discussion by Bonkles** [facebook/Rapid#481] [Announcements]
_(Edit: We released Rapid 2.0 on May 4th, 2023. Get it here: https://rapideditor.org/edit)_

Last week @Bonkles and I gave a talk at the [State of the Map US conference](https://2022.stateofthemap.us/) where we shared some details about our work on a _new_ WebGL-based renderer for RapiD built on the open source [PixiJS](https://pixijs.com/) game engine.  🎉 

The new WebGL-based renderer will allow RapiD to keep up with the increasing demands placed on it, as OpenStreetMap aims to map the world at greater levels of detail. Many places where RapiD was previously unusable now become responsive even at low zooms.

You can try out this new speedier RapiD at this link, where we will push updates sometimes:
https://mapwith.ai/rapid-sotm-2022

_This is an "alpha" release_ and there are for sure lots of bugs that need to be squashed.  We'll start tracking bugs and issues related to the new RapiD on this GitHub, with the label https://github.com/facebookincubator/RapiD/labels/webgl . If you find a bug, let us know!

We don't have a set deadline for when this work will replace the currently deployed RapiD, but we expect it will happen this year sometime, fingers crossed! 🤞 

Also - anyone interested in digging into the code more closely is encouraged to check out the `pixi_playground` branch.
https://github.com/facebookincubator/RapiD/tree/pixi_playground

Here are some links on Twitter that you can share too:  
https://twitter.com/MapWithAI/status/1510028065824391168
https://twitter.com/bhousel/status/1510053741025656835

And a short video showing the difference between the old and new renderer.
https://user-images.githubusercontent.com/38784/162043618-d3281c9a-5344-4e88-874c-b550f92684b5.mov

(We'll post a link to our talk whenever it is available)

Finally, please to reach out to us (Ben Clark, Bryan Housel) on [OSM-US Slack](https://slack.openstreetmap.us/) if you want to discuss more.  
We're always happy to have a conversation!

I'd also like to especially thank @sketchpunk, @vanreece, @danil-kirsanov, and the folks at @PixiJS team whose patience and work made this possible!  🙌

**Comment by simonschaufi** [comment]
Hello, I would like to test https://rapideditor.org/rapid-sotm-2022 but I can't login. The whole process doesn't complete. The callback from OSM is probably wrong because it just opens a landing page but doesn't actually log me in.

**Reply by tordans** [reply]
This was merged to main a while ago AFAIK

**Reply by Bonkles** [reply]
Tobias is correct! I will update the note to call this out. We released rapid v2 on May 4th of this year- it's now the editor you get when you visit rapideditor.org/edit. 

Expect a new github discussion post soon with our next steps and roadmap!
